src/Promenade.Controllers/HomeController.cs

```diff
@@ -1,21 +1,362 @@
+                                longitude = result.geometry.location.lng;
+
+                                Location viewModel = LibraryLookup(latitude, longitude);
+                                viewModel.FormattedAddress = result.formatted_address;
+                                if (mobile == 1)
+                                {
+                                    viewModel.ShowLocation = true;
+                                }
+                                return View("Locations", viewModel);
+                            }
+                            else
+                            {
+                                _logger.LogError($"No geocoding results for address \"{address}\"");
+                                TempData["AlertDanger"] = $"Unable to locate address <strong>\"{address}\"</strong>.";
+                            }
+                        }
+                        catch (HttpRequestException ex)
+                        {
+                            _logger.LogCritical(ex, $"Google API error: {ex.Message}");
+                            TempData["AlertDanger"] = "An error occured, please try again later.";
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger.LogCritical(ex, ex.Message);
+                            TempData["AlertDanger"] = "An error occured, please try again later.";
+                        }
+                    }
+                }
+                return View("Locations");
+            }
+            else
+            {
+                var viewModel = new Location();
+                viewModel.CloseLocations = locations.OrderBy( c => c.Name).ToList();
+                if (mobile==1)
+                {
+                    viewModel.ShowLocation = true;
+                }
+                return View("Locations", viewModel);
+            }
+        }
+
+        [HttpGet("[action]/{locationStub}")]
+        [HttpGet("[action]/{locationStub}/{featureStub}")]
+        public IActionResult Locations(string locationStub, string featureStub)
+        {
+            Location[] locations = {  };
+
+
+            Feature[] features = {  };
+
+
+            LocationFeature[] locationsFeat = {  };
+
+            var group = new Group { Id = 1, GroupType="distance" };
+
+
+            LocationGroup[] locagroups = { };
+
+
+            if (string.IsNullOrEmpty(locationStub))
+            {
+
+                return View("Locations",locations);
+            }
+            else if (string.IsNullOrEmpty(featureStub))
+            {
+                var locationViewModel = new LocationViewModel();
+
+                foreach (var location in locations)
+                {
+                    if (location.Stub == locationStub)
+                    {
+                        locationViewModel.Location = location;
+                        var featlist = new List<LocationsFeaturesViewModel>();
```

Comment:
Rename to `featureList`

Reply:
The variable declaration still needs to be renamed

Reply:
Fixed